StreamChatClient_v3/ChatClient.swift

```diff
@@ -42,106 +42,78 @@ public typealias ChatClient = Client<DefaultDataTypes>
     private(set) lazy var webSocketClient: WebSocketClient = {
-        let jsonParameter = WebSocketPayload<ExtraData>(user: self.currentUser, token: token)
-        
-        var urlComponents = URLComponents()
-        urlComponents.scheme = baseURL.wsURL.scheme
-        urlComponents.host = baseURL.wsURL.host
-        urlComponents.path = baseURL.wsURL.path.appending("connect")
-        urlComponents.queryItems = [URLQueryItem(name: "api_key", value: config.apiKey.apiKeyString)]
-        
-//      if user.isAnonymous {
-//          urlComponents.queryItems?.append(URLQueryItem(name: "stream-auth-type", value: "anonymous"))
-//      } else {
-        urlComponents.queryItems?.append(URLQueryItem(name: "authorization", value: token))
-        urlComponents.queryItems?.append(URLQueryItem(name: "stream-auth-type", value: "jwt"))
-        //      }
-        
-        let jsonData = try! JSONEncoder.default.encode(jsonParameter)
-        
-        if let jsonString = String(data: jsonData, encoding: .utf8) {
-            urlComponents.queryItems?.append(URLQueryItem(name: "json", value: jsonString))
-        } else {
-            //          logger?.log("❌ Can't create a JSON parameter string from the json: \(jsonParameter)", level: .error)
-        }
-        
-        guard let url = urlComponents.url else {
-            fatalError()
-            //          logger?.log("❌ Bad URL: \(urlComponents)", level: .error)
-            //          throw ClientError.invalidURL(urlComponents.description)
-        }
-        
-        var request = URLRequest(url: url)
-        request.allHTTPHeaderFields = authHeaders(token: token)
-        
-        //      let callbackQueue = DispatchQueue(label: "io.getstream.Chat.WebSocket", qos: .userInitiated)
-        //      let webSocketOptions: WebSocketOptions = [] // = stayConnectedInBackground ? WebSocketOptions.stayConnectedInBackground : []
-        //      let webSocketProvider = defaultWebSocketProviderType.init(request: request, callbackQueue: callbackQueue)
-        
+        // Set up event middlewares
         let middlewares: [EventMiddleware] = [
             // TODO: Add more middlewares
             HealthCheckFilter()
         ]
         
-        let wsClient = WebSocketClient(urlRequest: request,
-                                       eventDecoder: EventDecoder<ExtraData>(),
-                                       eventMiddlewares: middlewares)
-        return wsClient
+        // Create a connection request
+        let socketPayload = WebSocketPayload<ExtraData>(user: self.currentUser, token: token)
+        let webSocketEndpoint = Endpoint<Data>(path: "connect",
+                                               method: .get,
+                                               queryItems: nil,
+                                               requiresConnectionId: false,
+                                               body: ["json": socketPayload])
```

Comment:
Can you tell me why here `<Data>` is a response type?

Reply:
Oh, I forgot it there, this should be `EmptyResponse` or something like that.